src/main/java/reactor/ipc/netty/ReactorNetty.java

```diff
@@ -112,4 +303,27 @@ protected void channelIdle(ChannelHandlerContext ctx,
 	}
 
 	static final Object TERMINATED = new TerminatedHandlerEvent();
+	static final Logger log        = Loggers.getLogger(ReactorNetty.class);
+
+	/**
+	 * A handler that can be used to extract {@link ByteBuf} out of {@link ByteBufHolder},
+	 * optionally also outputting additional messages
+	 *
+	 * @author Stephane Maldini
+	 * @author Simon Baslé
+	 */
+	@ChannelHandler.Sharable
+	static final class ExtractorHandler extends ChannelInboundHandlerAdapter {
+
+		final BiConsumer<? super ChannelHandlerContext, Object> extractor;
+
+		ExtractorHandler(BiConsumer<? super ChannelHandlerContext, Object> extractor) {
+			this.extractor = Objects.requireNonNull(extractor, "extractor");
+		}
+
+		@Override
+		public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
+			extractor.accept(ctx, msg);
```

Comment:
I wouldn't have done it like that, but rather as a `Function<Object, Object>` that would translate inbound objects. Have it return null or the same instance to signal pass-through. Let the `ExtractorHandler` do the job of ctx.channelReadForward + the http specific version detect that a `LastHttpContent` has been converted (in which case we forward both the translation and `EMPTY_LAST_HTTP_CONTENT`.

This would lift the burden of not forgetting that last step from the user's shoulders...